client/src/entrypoints/contrib/typed_table_block/typed_table_block.js

```diff
@@ -268,13 +271,19 @@ export class TypedTableBlock {
     column.deletedInput.value = '';
     this.deletedFieldsContainer.appendChild(column.deletedInput);
 
-    const prependColumnButton = $(`<button type="button"
-      class="button button-secondary button-small button--icon text-replace prepend-column"
-      aria-label="${h(this.blockDef.meta.strings.INSERT_COLUMN)}"
-      title="${h(this.blockDef.meta.strings.INSERT_COLUMN)}">
-        <svg class="icon icon-plus icon" aria-hidden="true"><use href="#icon-plus"></use></svg>
-      </button>`);
+    const prependColumnButton = $(`<button type="button"></button>`)
+      .html(
+        `<svg class="icon icon-plus icon" aria-hidden="true"><use href="#icon-plus"></use></svg>`,
+      )
+      .addClass(
+        'button button-secondary button-small button--icon text-replace prepend-column',
+      )
+      .attr('aria-label', h(this.blockDef.meta.strings.INSERT_COLUMN))
+      .attr('aria-expanded', 'false')
+      .attr('title', h(this.blockDef.meta.strings.INSERT_COLUMN));
```

Comment:
Just wondering why it's necessary to break all of these attributes out onto their own lines, rather than just adding `aria-expanded="false"` to the template string?

Reply:
It is not necessary, we could directly add `aria-expanded="false"` to the template string. However, this approach makes it more readable and is also used elsewhere in the code (e.g. appendColumnButton), therefore, I decided to modify it that way.

appendColumnButton:
https://github.com/wagtail/wagtail/blob/286d7cbd6b581c5cc3db80b5833db8922969b5bc/client/src/entrypoints/contrib/typed_table_block/typed_table_block.js#L326-L334

Reply:
I don't really see how it's more readable to have a separate `attr` to add each attribute rather than just having them on the initial HTML. `appendColumnButton` does it that way because it's modifying the style of a button that already exists - the 'Add column' button starts out as a text-based button and switches to the plus icon after the first column has been added. Everything else in the code ([`deleteColumnButton`](https://github.com/wagtail/wagtail/blob/286d7cbd6b581c5cc3db80b5833db8922969b5bc/client/src/entrypoints/contrib/typed_table_block/typed_table_block.js#L294-L299), [`prependRowButton`](https://github.com/wagtail/wagtail/blob/286d7cbd6b581c5cc3db80b5833db8922969b5bc/client/src/entrypoints/contrib/typed_table_block/typed_table_block.js#L389-L394), [`deleteRowButton`](https://github.com/wagtail/wagtail/blob/286d7cbd6b581c5cc3db80b5833db8922969b5bc/client/src/entrypoints/contrib/typed_table_block/typed_table_block.js#L425-L430)) is using template strings.

(incidentally - keeping the `h()` function call would be incorrect when using `.attr`. It's necessary when writing a string of HTML, to ensure that any special characters such as `"` are escaped rather than interpreted as the end of the string - but that situation doesn't arise when assigning the attribute directly, and would just result in end-users seeing the escaped version.)

Please change this back to the template string version.

Reply:
@gasman Thanks for the above explanation and I forgot to mention one thing in `appendColumnButton`, `aria-label ` & `addClass` are being assigned twice and i also forgot to remove `aria-expanded` in clear method,

https://github.com/ayaan-qadri/wagtail/blob/50453d0a2579c9c6e8385749a640545c42c58f9e/client/src/entrypoints/contrib/typed_table_block/typed_table_block.js#L218

I have added commit for this with updated and passed unit test in this branch (https://github.com/ayaan-qadri/wagtail/commit/e2b60d6f3c7ae956e70cd74583365e0176c12b6e - you can cherry-pick if the code is fine or lemme know if i should raise a PR) 

![image](https://github.com/user-attachments/assets/92ea44c7-bdd3-4ad6-8fd2-8899e2ccfe7c)

Reply:
Thanks! Cherry-picking that fix now. (I think we want to set `aria-expanded` to false rather than removing it entirely though...)

Reply:
yeaaa, that is true removing it completely does not make sense, in fact, it is good to keep as it was, because in very rare case where there is only one column added and if `add-column-menu` is opened and delete the column then it will make it false even though it is expanded

E.g:

https://github.com/user-attachments/assets/9f377ea1-a12c-4cf4-a7a9-87fc68adaeee

Reply:
Hmm, true - OK, I'll revert that then!

Reply:
Thanks for the quick reply !, I have update on branch (it has not needed to change snapshot) as well https://github.com/ayaan-qadri/wagtail/commit/20db83852182e58f6a55466a392f054902b39009 if you want to cherry pick:
![image](https://github.com/user-attachments/assets/22106c20-01c7-405f-9da2-14b60fdc3489)